src/container.ts

```diff
@@ -497,13 +499,33 @@ export function createContainer<T extends object = any, U extends object = any>(
         throw new AwilixResolutionError(name, resolutionStack)
       }
 
-      // Pushes the currently-resolving module name onto the stack
-      resolutionStack.push(name)
+      const lifetime = resolver.lifetime || Lifetime.TRANSIENT
+
+      // if any of the parents have a shorter lifetime than the one requested,
+      // and the container is configured to do so, throw an error.
+      if (options?.errorOnShorterLivedDependencies) {
+        const maybeLongerLifetimeParentIndex = resolutionStack.findIndex(
+          ({ lifetime: parentLifetime }) =>
+            isLifetimeLonger(parentLifetime, lifetime),
+        )
+        if (maybeLongerLifetimeParentIndex > -1) {
+          throw new AwilixResolutionError(
+            name,
+            resolutionStack,
+            `Dependency '${name.toString()}' has a shorter lifetime than its parent: '${resolutionStack[
```

Comment:
If I am not mistaken, "parent" implies a direct relationship. Perhaps "than one of its ancestors" would be more accurate?

```suggestion
            `Dependency '${name.toString()}' has a shorter lifetime than one of its ancestors: '${resolutionStack[
```